src/main/java/uk/ac/cam/cl/dtg/teaching/pottery/controllers/RepoController.java

```diff
@@ -44,49 +44,43 @@
 import uk.ac.cam.cl.dtg.teaching.pottery.task.Task;
 import uk.ac.cam.cl.dtg.teaching.pottery.task.TaskCopy;
 import uk.ac.cam.cl.dtg.teaching.pottery.task.TaskIndex;
+import uk.ac.cam.cl.dtg.teaching.pottery.worker.Job;
+import uk.ac.cam.cl.dtg.teaching.pottery.worker.Worker;
 
 public class RepoController implements uk.ac.cam.cl.dtg.teaching.pottery.api.RepoController {
 
   protected static final Logger LOG = LoggerFactory.getLogger(RepoController.class);
   private RepoFactory repoFactory;
   private TaskIndex taskIndex;
+  private Worker worker;
 
   /** Create a new RepoController. */
   @Inject
-  public RepoController(RepoFactory repoFactory, TaskIndex taskIndex) {
+  public RepoController(RepoFactory repoFactory, TaskIndex taskIndex, Worker worker) {
     super();
     this.repoFactory = repoFactory;
     this.taskIndex = taskIndex;
+    this.worker = worker;
   }
 
   @Override
   public RepoInfoWithStatus makeRemoteRepo(
       String taskId,
-      Boolean usingTestingVersion,
-      Integer validityMinutes,
+      Boolean usingTestingVersionBoolean,
+      Integer validityMinutesInteger,
       String variant,
       String remote)
-      throws TaskNotFoundException, RepoExpiredException, RepoStorageException,
-          RetiredTaskException, RepoNotFoundException, TaskMissingVariantException {
+      throws TaskNotFoundException, RepoStorageException, RetiredTaskException,
+      RepoNotFoundException, TaskMissingVariantException {
     if (taskId == null) {
       throw new TaskNotFoundException("No taskId specified");
     }
-    if (usingTestingVersion == null) {
-      usingTestingVersion = false;
-    }
-    if (validityMinutes == null) {
-      validityMinutes = 60;
-    }
+    boolean usingTestingVersion = usingTestingVersionBoolean == null ? false
```

Comment:
Why did this need changing?

Reply:
If you assign to a parameter, it is no longer effectively final, so can't be used in the inner class below.

Reply:
Ah yes ok. Thanks